### DISCUSSION TITLE
Allow unpublishing versions which are not necessary for dependents?

### DISCUSSION THREAD

**Discussion by YounGoat** [npm/feedback#254] [Registry] [answered]
I own a package named  whose lastest version is 1.0.0 and it is depended by some other packages with version `"^1.0.0"`. One day I publish a new version 2.0.0 which contains something wrong. And, when I try to unpublish this version with:
```bash
npm unpublish <my-package-name>@2.0.0
```
A 405 error responsed:
```
npm ERR! Failed criteria:
npm ERR! has dependent packages in the registry
```
However, the version 2.0.0 has never been depended up to now. 

My advice is that __unpublishing package versions which are not necessary SHOULD BE permitted__.

**Comment by MylesBorins** [comment, marked as answer]
hey @YounGoat

I can understand the frustration here with not being able to unpublish that version that isn't specifically depended on. Our unpublish policy can be found [here](https://www.npmjs.com/policies/unpublish) and at this time we are not allowing the unpublishing of package or versions if the package itself has any dependents.

One option you do have is to deprecate the specific version 

`npm deprecate <package>@<version> "<message>"`

**Reply by YounGoat** [reply]
The `npm` command line tool has offered such suggestion. Although, in general, a version deprecated will not be depended any more, deprecation is still something different with unpublishing. 

Anyway, thanks for your reply.

**Reply by kroitor** [reply]
@MylesBorins deprecating a package version does not clean up the space used by it, right? We're stuck in a situation when we can't publish new package versions cause we've ran out of registry space quota, and we can't unpublish the old versions either, due to errors I've posted here: https://github.com/npm/feedback/discussions/785#discussioncomment-4783088
